useful_resources/sample_code/ngpvan_sample_list.py

```diff
@@ -0,0 +1,73 @@
+# ### METADATA
+
+# Connectors: VAN
+# Description: Creates a new saved list from a random sample of an existing saved list in VAN
+
+# ### CONFIGURATION
+
+# Set the configuration variables below or set environmental variables of the same name and leave
+# these with empty strings.  We recommend using environmental variables if possible.
+
+config_vars = {
+    # VAN
+    "VAN_API_KEY": "",
+    "VAN_DB_NAME": "",  # One of MyVoters, MyCampaign, MyMembers, EveryAction
+    "VAN_FOLDER_ID": "",  # Can be found in the URL on a Folder page in the FolderID param
+    "VAN_SAVED_LIST_ID": "",
+    "VAN_SAMPLE_LIST_SIZE": "",  # Must be a string
+    "VAN_SAMPLE_LIST_NAME": "",
+    # AWS
+    "AWS_ACCESS_KEY_ID": "",
+    "AWS_SECRET_ACCESS_KEY": "",
+    "S3_BUCKET": "",  # Bucket that the sample list data should be uploaded to
+}
+
+
+# ### CODE
+
+from parsons import VAN  # noqa: E402
+from parsons import logger  # noqa: E402
+import os  # noqa: E402
+import random  # noqa: E402
+
+# Setup
+
+# If variables specified above, sets them as environmental variables
+for name, value in config_vars.items():
+    if value.strip() != "":
+        os.environ[name] = value
+
+van = VAN(db=os.environ["VAN_DB_NAME"])
+
+# Get details on the saved list we're pulling from and the folder we're uploading to
+saved_list = van.get_saved_list(os.environ["VAN_SAVED_LIST_ID"])
+folder = van.get_folder(os.environ["VAN_FOLDER_ID"])
+
+saved_list_download = van.download_saved_list(os.environ["VAN_SAVED_LIST_ID"])
```

Comment:
A note here that using this method requires certain non-default permissions be turned on for an API key. I _believe_ all Parsons type VAN API keys have this turned on, but may be worth a comment to reach out to apidevelopers@ngpvan.com if struggling to get this to work.

Reply:
Just added a comment in the latest commit